src/control/Control.cpp

```diff
@@ -2009,10 +2009,18 @@ auto Control::openFile(fs::path filepath, int scrollToPage, bool forceOpen) -> b
         !loadHandler.getMissingPdfFilename().empty()) {
         // give the user a second chance to select a new PDF filepath, or to discard the PDF
 
-        GtkWidget* dialog = gtk_message_dialog_new(
-                getGtkWindow(), GTK_DIALOG_MODAL, GTK_MESSAGE_QUESTION, GTK_BUTTONS_NONE, "%s",
-                loadHandler.isAttachedPdfMissing() ? _("The attached background PDF could not be found.") :
-                                                     _("The background PDF could not be found."));
+        string missingFilePath = loadHandler.getMissingPdfFilename();
+        string msg1 =
+                FS(_F("The attached background file {1} could not be found. It might have been moved, renamed or "
+                      "deleted.\nIt was last seen at: {2}") %
+                   fs::path(missingFilePath).filename().string() % fs::path(missingFilePath).parent_path().string());
```

Comment:
To avoid the duplication of `fs::path`, I would declare `missingFilePath` to be `fs::path` instead of `string`.